dev_docs/tutorials/data/search.mdx

```diff
@@ -0,0 +1,490 @@
+
+  const searchSource = await data.search.searchSource.create();
+
+  searchSource
+    .setField('index', indexPattern)
+    .setField('filter', filters)
+    .setField('query', query)
+    .setField('fields', selectedFields.length ? selectedFields.map((f) => f.name) : ['*'])
+    .setField('aggs', getAggsDsl());
+    
+  searchSource.fetch$().subscribe({
+    next: () => {},
+    error: () => {},
+  });
+}
+```
+
+### Partial Results
+
+When searching using an `async` strategy (such as async DSL and async EQL), the search service will stream back partial results.   
+
+While you may ignore them and wait for the final result before rendering, you may also use those partial results to create a more interactive experience for your users. It is highly advised however, to make sure users are aware that the results they are seeing are partial.
+
+```ts
+// Handling partial results
+data.search.search(req).subscribe({
+  next: (result) => {
+    if (isCompleteResponse(res)) {
+      renderFinalResult(res);
+    } else if (isPartialResponse(res)) {
+      renderPartialResult(res);
+    }
+  },
+})
+
+// Skipping partial results
+const finalResult = await data.search.search(req).toPromise();
+```
+
+### Search Sessions
+
+A `Search Session` is a higher level concept, than search. It is used to describe a grouping of one or more async search requests with additional context. 
+
+`Search Sessions` come in handy when you want to enable a user to run something asynchronously (for example, a dashboard over a very long period of time), and then restore the results quickly at a later time, as the `Search Service` transparently fetches results from the `.async-search` index, instead of running each request again.
+
+Internally, any search run within the a `Search Session` is saved into a `Saved Object`, allowing `Kibana` to manage their lifecycle. Most `Saved Objects` are deleted automatically after a short period of time, but if a user chooses to save the `Search Session`, the `Saved Object` is persisted, so that results can be restored in a later time.
+
+Stored `Search Sessions` are listed in the management application, under `Kibana > Search Sessions`, making it easy to find, manage and restore them.
+
+As a developer, there might be two different, but common, use cases you might encounter:
+
+ * Running a search inside an existing Search Session
+ * Supporting Search Sessions in your application
+
+#### Running a search inside an existing Search Session
+
+For the sake of this example, lets assume you are implementing a new type of `Embeddable` that will be shown on dashboards. The same principle applies however, to any `search` requests you may be running, as long as the application you are running inside is managing an active session.
+
+Since the Dashboard application is already managing a `Search Session`, all you need to do is pass down the `searchSessionId` argument to any `search` call (this applies to both the low and high level search APIs). 
```

Comment:
```suggestion
Because the Dashboard application is already managing a search session, all you need to do is pass down the `searchSessionId` argument to any `search` call. This applies to both the low and high level search APIs. 
```